### ISSUE TITLE
Support for Array types

### ISSUE THREAD

**Issue by mannol** [RobinBuschmann/sequelize-typescript#594]
If you define a column like:
```ts
@Column(DataType.ARRAY(DataType.STRING))
myCol: string[];
```

You'll get the error saying: 
```
error TS2559: Type 'ArrayDataType<AbstractDataTypeConstructor>' has no properties in common with type 'Partial<ModelAttributeColumnOptions>'.
```

**Comment by devoto13** [comment]
Seems to be a regression in one of the latest releases. 

You can workaround it it by doing `@Column({ type: DataType.ARRAY(DataType.STRING) })` in the meantime.

**Comment by RobinBuschmann** [comment]
@mannol Thanks for bringing this up. Does this happen with sequelize-typescript@canary?

**Comment by mannol** [comment]
@RobinBuschmann no problem. Thanks for this library!

On your question, I believe I installed the lib via `yarn add sequelize-typescript@canary` like 2 weeks ago. `package.json` shows I'm using version `"^1.0.0-alpha.9` of this lib with sequelize version `^5.2.9`.

**Comment by RobinBuschmann** [comment]
@mannol Thanks. I wanted to make sure, that @devoto13's fix solves the issue. Since he made one for sequelize-typescript@1.

**Comment by mannol** [comment]
@RobinBuschmann sweet! Looks promising :+1:

**Comment by RobinBuschmann** [comment]
Closing this, since it was fixed with `sequelize-typescript@1.0.0-beta.1`

**Comment by ihoryan** [comment]
I'm trying to define array property and getting the error.
`\app\node_modules\sequelize-typescript\dist\model\shared\model-service.js:63
    throw new Error(`Specified type of property '${propertyName}'
          ^
source-map-support.js:419
Error: Specified type of property 'permissions'
            cannot be automatically resolved to a sequelize data type. Please
            define the data type manually`

sequelize-typescript@6.7.0
